ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -92,20 +93,30 @@
     this.applicationId = requireNonNull(applicationId, "applicationId");
   }
 
+  public List<KsqlPartitionLocation> locate(final List<KsqlKey> keys,
```

Comment:
why are we introducing this method? I feel like `isRangeScan` should always be passed in to maintain correctness. If it's just for testing, let's just update the tests. And speaking of... this change is missing test coverage in `KsLocatorTest` 😉

Reply:
Fair point.